tests/unit/test_charm.py

```diff
@@ -147,16 +171,42 @@ def test_relation_joined_when_leader_and_secrets(mock_juju_env):
     data = harness.model.get_relation("smtp").data[harness.model.app]
     assert data["host"] == host
     assert data["port"] == str(port)
-    assert "password" not in data
     assert data["password_id"] is not None
 
 
 @patch.object(ops.JujuVersion, "from_environ")
-def test_relation_joined_when_leader_and_no_secrets(mock_juju_env):
+def test_relation_joined_when_secrets_enabled_doesnt_populate_password(mock_juju_env):
+    """
+    arrange: set up a charm with valid configuration mimicking Juju 3 and leadership for the unit.
+    act: add an smtp relation.
+    assert: the smtp relation does not populate with the password.
+    """
+    mock_juju_env.return_value = MagicMock(has_secrets=True)
+    harness = Harness(SmtpIntegratorOperatorCharm)
+    harness.set_leader(True)
+    host = "smtp.example"
+    port = 25
+    password = "somepassword"  # nosec
+    harness.update_config(
```

Comment:
do you think this (the charm's config dict) can be defined in a fixture?

Reply:
Made a bit of refactoring